plugins/obs-webrtc/webrtc/src/ffi/output.rs

```diff
@@ -0,0 +1,204 @@
+use crate::output::{EncodedPacket, EncodedPacketType, OutputStream, OutputStreamError};
+use anyhow::Result;
+use log::{error, info};
+use std::{
+    os::raw::{c_char, c_void},
+    slice,
+    time::Duration,
+};
+use tokio::runtime::Runtime;
+
+pub struct OBSWebRTCWHIPOutput {
+    stream: OutputStream,
+    runtime: Runtime,
+}
+/// cbindgen:prefix-with-name
+#[repr(C)]
+pub enum OBSWebRTCWHIPOutputError {
+    ConnectFailed,
+    NetworkError,
+}
+
+impl From<OutputStreamError> for OBSWebRTCWHIPOutputError {
+    fn from(ose: OutputStreamError) -> Self {
+        match ose {
+            OutputStreamError::ConnectFailed => OBSWebRTCWHIPOutputError::ConnectFailed,
+            OutputStreamError::NetworkError => OBSWebRTCWHIPOutputError::NetworkError,
+        }
+    }
+}
+
+/// Create a new whip output in rust and leak the pointer to caller
+/// # Note
+/// You must call `obs_webrtc_whip_output_free` on the returned value
+#[no_mangle]
+pub extern "C" fn obs_webrtc_whip_output_new() -> *mut OBSWebRTCWHIPOutput {
+    (|| -> Result<*mut OBSWebRTCWHIPOutput> {
+        let runtime = tokio::runtime::Runtime::new()?;
+        let stream = runtime.block_on(async { OutputStream::new().await })?;
```

Comment:
```suggestion
        let stream = runtime.block_on(OutputStream::new())?;
```
should suffice

Reply:
Fixed in 79f98986dd7eff11023b1bd92b0a85c87cf1d1df